### ISSUE TITLE
Change Location Temporarily Closed text

### ISSUE THREAD

**Issue by Grateful-Viv** [code4recovery/spec#29]
Some of our online meetings that used to meet at a physical location, are now going back inside AND keeping the online meeting at the same time. Not as a hybrid meeting, but rather two separate meetings. 

For the online instance, it seems to be confusing people to see the text "location temporarily closed" since it's only this instance of this meeting that is not meeting in person. The location is open for other meetings.

Perhaps it could be changed to "Not meeting in person", or "This meeting is not meeting in person"(And take out the temporarily word). 

I would still have the address crossed and greyed out if type is TC regardless of whether the URL and/or phone field had data. 

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by joshreisner** [comment]
I've moved this issue to the spec. The TSML plugin and TSML-UI both adhere to this for the type names.

Some history on this type; it was created on Mar 12, 2020 originally phrased `Temporary Closure`.

It was changed on April 11 to `Location Temporarily Closed`.

There's a [proposal underway](https://github.com/code4recovery/spec/pull/19) to deprecate this type entirely and use a new `attendance` field for this, not have it be a meeting type at all.

Once that gets resolved, the plugin and TSML-UI will both follow whatever new rules are adopted.

**Comment by tech2serve** [comment]
Hey @Grateful-Viv,

For the use case your comment described in the first paragraph, I think many are making these two different meeting entries. 

Given this, for the online instance, I *think* you should be able to remove the Location Temporarily Closed tag from that meeting, no? You may to then change the meeting address to an approximate location (city and state)

If I’m understanding you correctly, your list should then show two meetings, one in person and one online.

Am I misunderstanding?

**Comment by Grateful-Viv** [comment]
@tech2serve 

We did make two meetings, one for the online which has TC as the type and the URL/Phone number entered. The other is the physical address meeting. But the complaint came through today the one that is online is confusing because it says location temporarily closed, rather than the meeting is not meeting in person.

I made a notation in the online meeting that there are two meetings, one at the location and this one that is online. See screenshots.

Yes, it's splitting hairs, but that's what I deal with often. 

![Desire to Stop Online](https://user-images.githubusercontent.com/62247421/120210933-58c39a00-c1fe-11eb-8b2b-d5259879009a.png)

![Desire to Stop Indoors](https://user-images.githubusercontent.com/62247421/120210980-67aa4c80-c1fe-11eb-9334-34ac840393b3.png)

**Comment by joshreisner** [comment]
Is the reason to make them two separate meetings so that they're not labeled "Hybrid?"

**Comment by Grateful-Viv** [comment]
> Is the reason to make them two separate meetings so that they're not labeled "Hybrid?"

No. In this example, the Desire to Stop meeting is NOT a hybrid meeting. It's two separate meetings. But for some reason, they are getting thrown off by the Location Temporarily Closed text from the online meeting. Hopefully, the note I added will clarify things.

We do have a couple of hybrid meetings, and those have TC type removed, proper URL/Phone fields filled out, along with with any notes in there regarding safety protocols.  

See for example:

https://sussexaa.org/meetings/friday-night-rehoboth/
https://sussexaa.org/meetings/love-service/

**Comment by tech2serve** [comment]
Why have the TC type applied to the online meeting? You’re not trying to tell me their regular meeting at (fill in the blank church) isn’t available any more. The meeting is a separate meeting and is in person.

You don’t need to have the TC type applied unless you’re trying to tell me the regular in-person meeting isn’t taking place.

Since you have two different meetings, that is moot.

**Comment by Grateful-Viv** [comment]
These are a separate meetings - each doing their own discussion, running at their own pace, at theoretically, different locations (one is online and the other is at a physical address). It seemed to make sense to add a second meeting to the already online meeting since they're different meetings.

Hybrid meetings are not separate. You can go there in person or you can go there online, but it's the same meeting.

If I were to follow your thinking, what would you do different if this was a hybrid meeting vs a separated physical meeting and online meeting?

**Comment by tech2serve** [comment]
> These are a separate meetings - each doing their own discussion, running at their own pace, at theoretically, different locations (one is online and the other is at a physical address). It seemed to make sense to add a second meeting to the already online meeting since they're different meetings.

This is why I don’t understand why you’d use the TC type for either of these meetings. They are separate and distinct, and operating as designed.

> 
> Hybrid meetings are not separate. You can go there in person or you can go there online, but it's the same meeting.

Correct.
> 
> If I were to follow your thinking, what would you do different if this was a hybrid meeting vs a separated physical meeting and online meeting?

Leave them as one entry in the database. Make sure the `conference_url` is filled out, and also make sure their is a complete physical address. And I would leave the “Location Temporarily Closed` **unchecked** as there are no restrictions for attending this meeting.

**Comment by Grateful-Viv** [comment]
>Leave them as one entry in the database. Make sure the conference_url is filled out, and also make sure their is a complete physical address. And I would leave the “Location Temporarily Closed` unchecked as there are no restrictions for attending this meeting

So you'd handle hybrid and the meeting that isn't hybrid the same way?

**Comment by tech2serve** [comment]
> > Leave them as one entry in the database. Make sure the conference_url is filled out, and also make sure their is a complete physical address. And I would leave the “Location Temporarily Closed` unchecked as there are no restrictions for attending this meeting
> 
> So you'd handle hybrid and the meeting that isn't hybrid the same way?

I’m not sure I understand the specifics leading up to this question.

What I am saying is that if a meeting is online only, it doesn’t require the TC type to be used *unless* there was a location associated with it that *isn’t* presently available. At that point, the meeting is no longer hybrid.

Now, let’s say that meeting location reopens and people can again attend in person. The admin should remove the TC type, and make sure the address is complete (not approximate) with the street, city and state. At that point, as long as the `conference_url` is still entered, the meeting *should* be handled as a hybrid. It will have one entry in your database.

In your situation, you have two distinct meetings and I’m not seeing any reason to use TC for either of them.

**Comment by Grateful-Viv** [comment]
So you would have two meetings listed? One has the conference URL info. The other has the physical address. Neither has TC checked? 

This conversation is getting hard to follow.

Maybe we could do a quick zoom meeting.

**Comment by tech2serve** [comment]
If the meetings are indeed two different and distinct meetings, then yes, I would have two entries in the database with unique names. 

To help differentiate, does your online and in-person meeting have the same chair person? Is there a camera in the physical room so that the people in the building can speak with, see and listen to someone connected virtually? If the answers are "no", then I'd say they are two distinct meetings, and there should be two separate meetings (with different names).

As far as entering them, the online meeting should have the `conference_url` and an approximate location (city and state). The in-person meeting should not have the `conference_url` and it should be populated with a complete address (street, city and state).

**Comment by Grateful-Viv** [comment]
Two different meetings. Two different chairs. Two different discussions. 

Not hybrid. No camera.

In-person meeting has physical location that is properly linked to map. No TC.

Online meeting has conference URL. Also has TC type enabled. I don't think that should come off. The text could be changed to say the meeting is not in-person, instead of the location temporarily closed. I know what it was originally, and people were confused and that's why it changed, but it would actually be less confusing to say the meeting is not meeting in-person, rather than the location is closed.

The only time I have TC off for an online meeting is if it's hybrid, which this is not.

**Comment by tech2serve** [comment]
> Two different meetings. Two different chairs. Two different discussions.
> 
> Not hybrid. No camera.
> 
> In-person meeting has physical location that is properly linked to map. No TC.
> 
> Online meeting has conference URL. Also has TC type enabled. I don't think that should come off. The text could be changed to say the meeting is not in-person, instead of the location temporarily closed. I know what it was originally, and people were confused and that's why it changed, but it would actually be less confusing to say the meeting is not meeting in-person, rather than the location is closed.
> 
> The only time I have TC off for an online meeting is if it's hybrid, which this is not.

All I can really say is that is not how we're designing it (which is based on a lot of discussion and feedback). I'll add a couple more thoughts.

The meeting list is primarily intended to be a snapshot in time, at this moment right now. It isn't meant for historical tracking. Just because an online meeting grew out of a physical location that was closed because of the pandemic doesn't mean its listing should show the meeting location as closed, unless that closure is temporary and you intend to reopen it, thus making it a hybrid meeting once that occurs. In the case you're describing, it sounds like this isn't the situation. I see no reason to continue to have TC associated with the online meeting.  

Think of it this way. Regardless if the label displays "location temporarily closed" or "temporarily suspended" (or something else), the intention of this metadata is to alert the user that if they show up to the meeting location because it was an in-person meeting yesterday (think more literally vice several months ago), they will find the doors locked or the meeting location otherwise unaccessible. This doesn't apply to a meeting that has moved completely and permanently online, and the reason for the inactive state should be described in the notes.

The entire TC type is (long term) going away, or at least changing. Some descriptor, such as "inactive", is probably the direction we're going. There are a number of reasons for this change, but the overall intent is to simplify the messaging to the newcomer that the meeting they're considering isn't actually available to them right at this moment. There is a pretty strong push to remove these from the active list entirely, but a full consensus hasn't been reached yet and we still understand the need for temporary closures stemming from events like building remodels, summer schools, seasonal flooding, etc. So we're maintaining support for that.

But with an online meeting that doesn't rely upon a physical address (short of an approximate location for affiliation), adding the TC type is more likely to confuse the newcomer. I don't see where it adds any value outside of the situation where the meeting will soon return to an in-person format at the physical location.

**Comment by Grateful-Viv** [comment]
A lot of these meetings once met at a physical location. In fact most of them did. 

We used TC to indicate the location closed. 

If it stayed closed and no online meeting occurred then as far as we’re concerned the meeting is inactive, which is something I’ve been asking for for a while as a flag. 

If the meeting became online, we still used TC and then entered the URL. Many of these groups have been paying rent all this time in order to hold their space. It’s just that they’re currently meeting online, but it’s been useful to see address displayed as greyed-out and crossed-out to let people know that there’s a placeholder for if and when it returns in person. 

This is completely different from meetings that got created as an online meeting only and never had a physical presence. In those cases we enter the county and the state and no physical street address. There’s no TC involved in this one either. 

TC was removed if the meeting became hybrid or if they were turned in person to the location. 

You’re suggesting now that I remove the TC and also remove the address?

**Comment by tech2serve** [comment]
> 
> You’re suggesting now that I remove the TC and also remove the address from all online meetings?

In short, yes, as long as the meeting is purely online or is hybrid (and the meeting venue has reopened). If the meeting is online but still has a physical venue that you expect to reopen soon, then it is reasonable to leave the TC type checked.

In our experience, most areas are rapidly reopening meetings which means they're removing "Location Temporarily Closed". For those meetings that also have the online details filled out, these meetings should be considered hybrids, unless the address is approximate (i.e., city and state only) If the meeting is shifting from online to in-person, the conference details should be removed, and the meeting should be treated as once again in-person if the address is complete (i.e., street, city and state). 

The next couple paragraphs are my attempt to further share our thinking outside of our group for the purpose of practicing how well we can communicate our intentions :-) I'm hoping this makes sense to you (not necessarily if you agree with it, just simply if you can follow our logic):

We're still working with this for TSML v3.11. As you're aware I think, we're adding an `attendance_option` to better characterize meetings as "in_person", "hybrid" or "online". This field also has the option of reflecting the meeting as temporarily inactive, but we're still working on the best way to do this automatically in order to simplify the UX for the admin.

<img width="782" alt="image" src="https://user-images.githubusercontent.com/31904316/120240996-883acc80-c227-11eb-9aae-4a03f1337044.png">


 My current thinking is that if the admin answers "no" to the question about the meeting being in-person, and conference url is filled out, then we're left with two scenarios: 

1) A specific street address is provided. In this case I think the plugin should still treat the meeting location as temporarily closed, and the meeting should be flagged as such.
2) Only an approximate location is provided. In this case, the plugin should treat the meeting as purely online.

Perhaps [this description](https://github.com/code4recovery/tsml-ui#how-are-metatypes-like-active-and-online-calculated) will help explain the direction we're going?

**Comment by Grateful-Viv** [comment]
>If the meeting is online but still has a physical venue that you expect to reopen soon, then it is reasonable to leave the TC type checked.

Reopen soon is very subjective. Some are jumping at it, and others are being very cautious. I think we'll have a better picture by mid-summer. In the meantime, I'd hate to remove the TC for them.

As far as meetings that are either hybrid or purely online, there is no TC type checked.

As far as your scenario described with "no" to the meeting being in-person and conference URL filled out and a specific street address filled out, maybe consider that instead of treating this as "location temporarily closed" instead to  "meeting is not meeting in-person".  The reason for this is the location could be open for other meetings, but this particular meeting is not meeting in person.

**Comment by tech2serve** [comment]
> > If the meeting is online but still has a physical venue that you expect to reopen soon, then it is reasonable to leave the TC type checked.
> 
> Reopen soon is very subjective. Some are jumping at it, and others are being very cautious. I think we'll have a better picture by mid-summer. In the meantime, I'd hate to remove the TC for them.

Agreed. 
> 
> As far as meetings that are either hybrid or purely online, there is no TC type checked.

Yes. And that is the point I was trying to make earlier. If you have a meeting that is online, and there is no intention of it having a corresponding physical meeting, leave the TC off. It doesn't add any value. 

I suspect there will be meetings that had been in person prior to the pandemic. They shuttered last year (TC'd), then went online (ONL and TC'd), and now the group has decided to skip reopening the physical meeting and instead will just keep the group online. In this case, it makes total sense to me to remove the street address and the TC type, and make the meeting online only.
> 
> As far as your scenario described with "no" to the meeting being in-person and conference URL filled out and a specific street address filled out, maybe consider that instead of treating this as "location temporarily closed" instead to "meeting is not meeting in-person". The reason for this is the location could be open for other meetings, but this particular meeting is not meeting in person.

I think we're probably going with "inactive" for the label displayed to the end user for just this reason.

**Comment by Grateful-Viv** [comment]
> Yes. And that is the point I was trying to make earlier. If you have a meeting that is online, and there is no intention of it having a corresponding physical meeting, leave the TC off. It doesn't add any value.

Agreed. And I got that point from the beginning. 

> I suspect there will be meetings that had been in person prior to the pandemic. They shuttered last year (TC'd), then went online (ONL and TC'd), and now the group has decided to skip reopening the physical meeting and instead will just keep the group online. In this case, it makes total sense to me to remove the street address and the TC type, and make the meeting online only.

That makes total sense. I know of several that seem to like their zoom more than going in-person.

> I think we're probably going with "inactive" for the label displayed to the end user for just this reason.

Not sure I follow. Inactive would only apply to suspended meetings that are not meeting anywhere as far as we’ve been told. Is that what you mean? Because my scenario was for the TC and online that used to meet someplace, but the meeting “temporarily” moved online. I know when this all started we didn’t think temporarily would be over a year.

**Comment by tech2serve** [comment]
> Not sure I follow. Inactive would only apply to suspended meetings that are not meeting anywhere as far as we’ve been told. Is that what you mean? Because my scenario was for the TC and online that used to meet someplace, but the meeting “temporarily” moved online. I know when this all started we didn’t think temporarily would be over a year.

We're still working through how to keep this simple for the admin while still keeping it comprehensible.

More to follow.
